File: scala/src/test/scala/com/pduda/field/ImagineThisIsAClientInADifferentRepositoryTest.scala
package com.pduda.field

import org.junit.Assert

class ImagineThisIsAClientInADifferentRepositoryTest extends munit.FunSuite {
  test("single item: number of products in the cart") {
    val client = new ImagineThisIsAClientInADifferentRepository
    Assert.assertEquals("Total price is 50 euro", client.formattedTotalPrice(50))
  }
}

File: Java/src/test/java/com/pduda/field/ImagineThisIsAClientInADifferentRepositoryTest.java
package com.pduda.field;

import org.junit.Assert;
import org.junit.Test;

public class ImagineThisIsAClientInADifferentRepositoryTest {

    /**
     * Why: Verify that the client correctly formats the total price for a given
     * input.
     */
    @Test
    public void singleItemNumberOfProductsInTheCart() throws Exception {
        ImagineThisIsAClientInADifferentRepository client = new ImagineThisIsAClientInADifferentRepository();

        Assert.assertEquals("Total price is 50 euro", client.formattedTotalPrice(50));
    }
}
